### PR TITLE
[gff_tigrinya_ethiopia] Tigrinya Ethiopia Keyboard

### PR DESCRIPTION
**Pull request by MakaraSok**
(update branch name)

### CODE REVIEW

**Comment by dyacob** [comment]
This PR would be ready following the processing of the dependent PR: https://github.com/MakaraSok/keyboards/pull/4

**Comment by LornaSIL** [comment]
This keyboard works fine. I'd like to see some of the same changes to the documentation as suggested for the Tigrinya (Eritrean) keyboard.

*Keyboard documentation:*
Under "Typing all languages". I believe you could keep the first paragraph including "However, on the occasion that you need to type Ethiopic letters outside of Tigrinya, you can do so with the “Unicode Keyboard”." Then tell them how to find the Unicode Keyboard for installing. Then delete all the other information about keyboarding because that should be in the documentation for the "Unicode Keyboard". Also, I think "Unicode Keyboard" should probably be a different name. Do you mean the Geez keyboard?

*welcome.htm*
I would like there to be significant review. The information on Keyman is out of date. The link to the ftp site is broken. There are no fonts being distributed with this keyboard ("The GNU FreeFont Project is responsible for the Free Serif font which is included with this package"). It's good to have the links to the fonts in there. It would be nice to update the link to Abyssinica SIL (https://software.sil.org/abyssinica/).

**Comment by dyacob** [comment]
@LornaSIL  Thanks for the review.  I need to understand the context of use for the manual and welcome.htm files in the planned packaging.  

In the "Power Pack" packagings that I was providing, there would be 2 keyboards included along with fonts.  One keyboard was language specific, the other was language neutral -this is what is referred to as the "Unicode Keyboard" in the manual (which also uses a Unicode icon).  So the manuals would always cover both keyboards.  This was more convenient (for myself and the user) than having two manuals. The character inventory that would be covered in the "Typing all languages" section for the "Unicode Keyboard", is also different for each language is it describes the delta from the target language.

The welcome.htm file mostly covered the font inventory and should have links updated.  Both the manual and welcome.htm I think make sense in the "Power Pack" context, but probably not in a stand alone context for the keyboards.  Depending on how the keyboards would be distributed, we may want to have separate "stand alone" documentation and the bundled "Power Pack" versions of the documentation files. I've understood that the intent is to still provide the power packs.

File: release/gff/gff_tigrinya_ethiopia/source/welcome.htm [hunk 1]
```diff
@@ -6,23 +6,20 @@
 <h1>The GFF Tigrinya Keyboard Package</h1>
 
 <p style="text-align: justify;">
-The contents of this package contain both free (OpenSource) and
-commercial software from various organizations. Please review the
+The contents of this package contain a free (OpenSource). Please review the
```
Comment (mcdurdin) [inline release/gff/gff_tigrinya_ethiopia/source/welcome.htm]: `The contents of this package are free and open source.`

File: release/gff/gff_tigrinya_ethiopia/source/welcome.htm [hunk 2]
```diff
@@ -6,23 +6,20 @@
...
 </p>
 
 
 <h1>Keyboards</h1>
 <p style="text-align: justify;">
-Two keyboards are provided with this package for Ethiopic script under the conventions of the Tigrinya language, and the full
-Ethiopic in a language neutral form. The keyboards are free software and you may share them freely with friends. The keyboards themselves
-may be downloaded independently from <a href="ftp://ftp.ethiopic.org/pub/keyboards/Keyman/7.0/">ftp://ftp.ethiopic.org/pub/keyboards/Keyman/7.0/</a>.
+This package is for Ethiopic script under the conventions of the Tigrinya language, and the full
+Ethiopic in a language neutral form. The keyboard is a free software and you may share it freely with friends.
```
Comment (mcdurdin) [inline release/gff/gff_tigrinya_ethiopia/source/welcome.htm]: `The keyboard is a free software` -> The keyboard is free software`

File: release/gff/gff_tigrinya_ethiopia/source/welcome.htm [hunk 3]
```diff
@@ -6,23 +6,20 @@
...
 </p>
 <p style="text-align: justify;">
-The software required to run the keyboards, "Keyman" is a commercial product developed by the <a href="http://tavultesoft.com/">Tavultesoft
-Proprietary Limited Company</a>. Keyman is free for a period of time before it must be purchased from Tavultesoft to continue functioning.
-See the <a href="http://tavultesoft.com">Tavultesoft homepage</a> for purchasing details.
+The software required to run the keyboards is "Keyman," a free and open source software.
```
Comment (mcdurdin) [inline release/gff/gff_tigrinya_ethiopia/source/welcome.htm]: `The software required to run the keyboards is "Keyman," which is free and open source.`

File: release/gff/gff_tigrinya_ethiopia/source/welcome.htm (hunk 1 above)
Comment (MakaraSok) [reply release/gff/gff_tigrinya_ethiopia/source/welcome.htm]: OK

File: release/gff/gff_tigrinya_ethiopia/source/welcome.htm (hunk 2 above)
Comment (MakaraSok) [reply release/gff/gff_tigrinya_ethiopia/source/welcome.htm]: OK

File: release/gff/gff_tigrinya_ethiopia/source/welcome.htm (hunk 3 above)
Comment (MakaraSok) [reply release/gff/gff_tigrinya_ethiopia/source/welcome.htm]: OK

**Comment by MakaraSok** [comment]
To investigate further:
- key 'u' outputs 'አ', but the correct one may be 'ኡ'
- keys 'A, Au, Ai, Aa, Aie, Ae, Ao' do not output characters the same as that of 'aaa, uu, ii, aa, iie, ee, oo'
- key ''' outputs two apostrophes ''''
- Ctrl + Alt + Y doesn't make the aforementioned icon appear.

(These are also commented in the pdf documentation.)

**Comment by dyacob** [comment]
I'll review and update the Tigirnya keyboards next, following the processing of the pull request for Blin which is similar:  https://github.com/MakaraSok/keyboards/pull/13

**Comment by LornaSIL** [comment]
@dyacob 
Minor: add copyright date to the welcome.htm (in the license section)
Bigger: ```??``` does not turn into the Tigrigna question mark.

Those are the only two things I see.

**Comment by MakaraSok** [comment]
@LornaSIL @dyacob fixed, PTAL.

**Comment by dyacob** [comment]
@MakaraSok and @LornaSIL , apologies that I couldn't have responded sooner.  A couple things, this will need a new PR merger.  The issue was with the documentation, not the keyboard.

@MakaraSok please merge: https://github.com/MakaraSok/keyboards/pull/31

which also includes the copyright date addition to the license section of the welcome.htm file that @LornaSIL requested (I'll need to add this to other keyboards accordingly).

@MakaraSok , can you also back out the changes in your last commit (.kmn , .kvk and .ico files)?  I'm having difficulty merging your upstream fork-branch back into mine.  Unfortunately, the .ico file gets broken each time Keyman rebuilds a keyboard, so care is needed not to overwrite the "good" .ico file (this issue is reported in https://github.com/keymanapp/keyman/issues/296).

thank you,

**Comment by MakaraSok** [comment]
@dyacob MakaraSok#31 has been merged.

> @MakaraSok , can you also back out the changes in your last commit (.kmn , .kvk and .ico files)? I'm having difficulty merging your upstream fork-branch back into mine. 

Could you please point to that specifically? Give me the commit name, and I'll remove it.

**Comment by dyacob** [comment]
@MakaraSok thanks for the merge, the commit I'm referring to would be this one:

https://github.com/MakaraSok/keyboards/commit/46c7c9a2793f2b008a0d0649048ff51ddca6f077

**Comment by dyacob** [comment]
Correct.  Its modern use is just found in Eritrea, a government paper (Hadas Eritrea) was using it regularly until recently.  Mixed preferences there though so I plan to make it a configure option shortly.

**Comment by LornaSIL** [comment]
LGTM

**Comment by MakaraSok** [comment]
@LornaSIL @dyacob it looks like the build agent is not working. It may take a while until @mcdurdin can look into this and get it back up.

File: release/gff/gff_tigrinya_ethiopia/source/gff_tigrinya_ethiopia.kps [hunk 4]
```diff
@@ -82,5 +58,15 @@
       <FileType>.ttf</FileType>
     </File>
   </Files>
+  <Keyboards>
+    <Keyboard>
+      <Name>ትግርኛ (Tigrinya)</Name>
+      <ID>gff_tigrinya_ethiopia</ID>
+      <Version>1.1</Version>
+      <Languages>
+        <Language ID="ti">ti</Language>
```
Comment (mcdurdin) [inline release/gff/gff_tigrinya_ethiopia/source/gff_tigrinya_ethiopia.kps]: Language name should be "Tigrigna (Ethiopia)", ID should be `ti-et`.

**Comment by LornaSIL** [comment]
https://keyman.com/keyboards/gff_tigrinya_ethiopia

**Pull request merged**